generators/python/core_utilities/sdk/query_encoder.py

```diff
@@ -0,0 +1,35 @@
+import httpx
+from typing import Any
+
+from .pydantic_utilities import pydantic_v1
+
+from collections import ChainMap
+
+
+# Flattens dicts to be of the form {"key[subkey][subkey2]": value} where value is not a dict
+def traverse_query_dict(dict_flat, key_prefix=None) -> dict:
+  result = {}
+  for k, v in dict_flat.items():
+    key = f'{key_prefix}[{k}]' if key_prefix is not None else k
+    if isinstance(v, dict):
+      result.update(traverse_query_dict(v, key))
+    else:
+      result[key] = v
+  return result
+
+
+def single_query_encoder(query_key: str, query_value: Any) -> dict:
```

Comment:
can we add a couple pytest unit tests for this function -- but othwerise it looks good

Reply:
some tests added